crates/common/mqtt_channel/src/session.rs

```diff
@@ -89,6 +86,7 @@ pub async fn clear_session(config: &Config) -> Result<(), MqttError> {
         }
     }
 
+    // Errors on disconnect are ignored, since having no impact on the session
```

Comment:
Is this also true for the broker? Does anything bad happen if a session is kept indefinitely by accident? 

While I don't think that not returning an error is bad, maybe the result should still be logged, so that in case there _might_ be a problem the information is not lost.

Reply:
The MQTT server protects itself with a [keep alive mechanism](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349237).